llama-index-integrations/readers/llama-index-readers-oracleai/llama_index/readers/oracleai/base.py

```diff
@@ -0,0 +1,457 @@
+                    sql = (
+                        "select dbms_vector_chain.utl_to_text(t."
+                        + self.colname
+                        + ", json('"
+                        + json.dumps(m_params)
+                        + "')) mdata, dbms_vector_chain.utl_to_text(t."
+                        + self.colname
+                        + ") text"
+                        + self.mdata_cols_sql
+                        + " from "
+                        + self.owner
+                        + "."
+                        + self.tablename
+                        + " t"
+                    )
+
+                    cursor.execute(sql)
+                    for row in cursor:
+                        metadata = {}
+
+                        if row is None:
+                            doc_id = OracleDocReader.generate_object_id(
+                                self.conn.username
+                                + "$"
+                                + self.owner
+                                + "$"
+                                + self.tablename
+                                + "$"
+                                + self.colname
+                            )
+                            metadata["_oid"] = doc_id
+                            results.append(Document(text="", metadata=metadata))
+                        else:
+                            if row[0] is not None:
+                                data = str(row[0])
+                                if data.startswith(("<!DOCTYPE html", "<HTML>")):
+                                    p = ParseOracleDocMetadata()
+                                    p.feed(data)
+                                    metadata = p.get_metadata()
+
+                            doc_id = OracleDocReader.generate_object_id(
+                                self.conn.username
+                                + "$"
+                                + self.owner
+                                + "$"
+                                + self.tablename
+                                + "$"
+                                + self.colname
+                                + "$"
+                                + str(row[2])
+                            )
+                            metadata["_oid"] = doc_id
+                            metadata["_rowid"] = row[2]
+
+                            # process projected metadata cols
+                            if self.mdata_cols is not None:
+                                ncols = len(self.mdata_cols)
+
+                            for i in range(ncols):
+                                if i == 0:
+                                    metadata["_rowid"] = row[i + 2]
+                                else:
+                                    metadata[self.mdata_cols[i]] = row[i + 2]
+
+                            if row[1] is None:
+                                results.append(Document(text="", metadata=metadata))
```

Comment:
Feels kinda weird to be creating documents without data, why is this the case?

Also if you want, you can set `Document(.., id_=doc_id)` instead

Reply:
Ideally, this should not happen. In case, doc is empty or null, we added:
Document(text="", metadata=metadata)

Here, metadata has the id along with other fields. So Document(.., id_=doc_id) case is covered.